### ISSUE TITLE
Spatial audio is glitchy

### ISSUE THREAD

**Issue by nobbele** [RustAudio/rodio#369]
```rust
fn main() {
    let (_stream, handle) = rodio::OutputStream::try_default().unwrap();
    let sink =
        rodio::SpatialSink::try_new(&handle, [-2.0, 0.0, 0.0], [1.0, 0.0, 0.0], [-1.0, 0.0, 0.0])
            .unwrap();

    let file = std::fs::File::open("beep.wav").unwrap();

    let source = rodio::Decoder::new(file).unwrap();
    sink.append(source);

    sink.sleep_until_end();
}
```
using this code (use any beep.wav file), nearing the end of the audio, it will start to play the sound in the other ear before switching back.

**Comment by DaforLynx** [comment]
Testing this code with a very short sound (less than half a second) over and over, this glitchiness is even more apparent. And when run in quick succession, the sound will occasionally swap sides for its duration.

**Comment by denizsincar29** [comment]
It is a very serious bug that rodio developers doesn't seam to fix. I also wrote an issue on this.

**Comment by denizsincar29** [comment]
upd. This issue is almost 1 year old! Dear developers. What are you doing for 1 year? Rodio is the most popular crate but has ridiculous bugs.

**Comment by est31** [comment]
@denizsincar29 bumping old threads every now and then is fine-ish, but the way you expressed your frustration is absolutely not. Software often has bugs for way longer than one year. This is a project ran by volunteers, and we are not here to read such comments.

**Comment by denizsincar29** [comment]
yep sorry. I didn't find a crate that would play 3d sound as easy as rodio, and I'm trying to make an audio game for blind people- (i'm also blind). audio games are based on 3d sound and I want to be concentrated on the game logic, not on the millions of lines of code with e.g. hrtf library. Rodio's spatial sink is perfect for that one, because it's easy to use. but buggy.
I know that i was a bit rood maybe...

**Comment by denizsincar29** [comment]
Hello! I updated rodio to the latest version but this bug doesn't seam to be fixed. Is there a way to workarround?
